File: 백준/Silver/10773. 제로/제로.cc
#include <iostream>
#include <stack>
using namespace std;

int main()
{
    int K;
    cin >> K;

    stack<int> num;

    for (int i = 0; i < K; i++)
    {
        int a;
        cin >> a;
        if (a != 0)
        {
            num.push(a);
        }
        else
        {
            num.pop();
        }
    }

    int sum = 0;
    while (!num.empty())
    {
        sum += num.top();
        num.pop();
    }

    cout << sum;
}

File: 백준/Silver/10773. 제로/제로.py
from collections import deque

K = int(input())

s = deque()
for _ in range(K):
    a = int(input())

    if a != 0:
        s.append(a)
    else:
        s.pop()

sum = 0
while s:
    sum += s.pop()

print(sum)
